### DISCUSSION TITLE
How to use borgbackup without a password?

### DISCUSSION THREAD

**Discussion by pmorch** [nextcloud/all-in-one#3013] [Questions] [answered]
If figured out that the borgbackup password is stored in `BORGBACKUP_PASSWORD` in `secrets` in `/mnt/docker-aio-config/data/configuration.json` in the container (or in the volume if you prefer). I know how to modify the password in both borgbackup and in that file. And that works.

But I'd prefer to not use a password for the borgbackup archive at all (one less thing that can go wrong during restore, and I use borgbackup already for other things without a password). I know how to remove the password from borgbackup using `sudo borg change-passphrase $storagedir/borg`, and then just hit enter for the new password. But nextcloud-aio doesn't seem to work without a password.

Is there any way I can use nextcloud-aio without a (aka with an empty) `BORGBACKUP_PASSWORD` ?

If I set the password to the empty string in the config file, I get this error message:

```
NOTICE: PHP message: Slim Application Error
Type: Exception
Code: 0
Message: The secret BORGBACKUP_PASSWORD is empty. Cannot substitute its value. Please check if it is defined in secrets of containers.json.
File: /var/www/docker-aio/php/src/Docker/DockerActionManager.php
```

If I remove the `BORGBACKUP_PASSWORD` from `secrets` in `/mnt/docker-aio-config/data/configuration.json`, a new one is created and so I get the error:

```
Performing backup...
Starting the backup...
passphrase supplied in BORG_PASSPHRASE, by BORG_PASSCOMMAND or via BORG_PASSPHRASE_FD is incorrect.
Deleting the failed backup archive...
passphrase supplied in BORG_PASSPHRASE, by BORG_PASSCOMMAND or via BORG_PASSPHRASE_FD is incorrect.
Backup failed!
You might want to check the backup integrity via the AIO interface.
```

**Comment by szaimen** [comment, marked as answer]
> How to use borgbackup without a password?

You cannot

**Reply by pmorch** [reply]
Ok. Thanks for the quick response!
